Buildings/Controls/OBC/CDL/Logical/ConstantDutyCycle.mo

```diff
@@ -0,0 +1,176 @@
+        Polygon(
+          points={{-80,88},{-88,66},{-72,66},{-80,88}},
+          lineColor={255,0,255},
+          fillColor={255,0,255},
+          fillPattern=FillPattern.Solid),
+        Line(
+          points={{-80,66},{-80,-92}},
+          color={255,0,255}),
+        Line(
+          points={{-90,-80},{72,-80}},
+          color={255,0,255}),
+        Polygon(
+          points={{90,-80},{68,-72},{68,-88},{90,-80}},
+          lineColor={255,0,255},
+          fillColor={255,0,255},
+          fillPattern=FillPattern.Solid),
+        Ellipse(
+          extent={{71,7},{85,-7}},
+          lineColor=DynamicSelect({235,235,235},
+            if y then
+              {0,255,0}
+            else
+              {235,235,235}),
+          fillColor=DynamicSelect({235,235,235},
+            if y then
+              {0,255,0}
+            else
+              {235,235,235}),
+          fillPattern=FillPattern.Solid),
+        Text(
+          extent={{-46,-12},{-14,-30}},
+          textColor={135,135,135},
+          textString="%period"),
+        Polygon(
+          points={{0,-30},{-8,-28},{-8,-32},{0,-30}},
+          fillColor={135,135,135},
+          fillPattern=FillPattern.Solid,
+          pattern=LinePattern.None,
+          lineColor={0,0,0}),
+        Line(
+          points={{-60,-30},{0,-30}},
+          color={135,135,135}),
+        Polygon(
+          points={{-60,-30},{-52,-28},{-52,-32},{-60,-30}},
+          fillColor={135,135,135},
+          fillPattern=FillPattern.Solid,
+          pattern=LinePattern.None,
+          lineColor={0,0,0}),
+        Line(points={{-80,0},{-60,0},{-60,40},{60,40}}, color={255,0,0}),
+        Line(points={{-80,-40},{-60,-40}}, color={0,0,0}),
+        Line(points={{-60,-40},{-30,-40},{-30,-80}}, color={0,0,0}),
+        Line(points={{0,-40},{30,-40},{30,-80}}, color={0,0,0}),
+        Line(points={{0,-40},{0,-80}}, color={0,0,0}),
+        Polygon(
+          points={{-60,-60},{-52,-58},{-52,-62},{-60,-60}},
+          fillColor={135,135,135},
+          fillPattern=FillPattern.Solid,
+          pattern=LinePattern.None,
+          lineColor={0,0,0}),
+        Line(
+          points={{-60,-60},{-30,-60}},
+          color={135,135,135}),
+        Polygon(
+          points={{-30,-60},{-38,-58},{-38,-62},{-30,-60}},
+          fillColor={135,135,135},
+          fillPattern=FillPattern.Solid,
+          pattern=LinePattern.None,
+          lineColor={0,0,0}),
+        Text(
+          extent={{-58,-44},{-34,-58}},
+          textColor={135,135,135},
+          textString="%width"),
+        Text(
+          extent={{-100,140},{100,100}},
+          textColor={0,0,255},
+          textString="%name")}),
+    Documentation(
+      info="<html>
+<p>
+Block that produces an output that cycles on and off according to the specified
+length of time for the cycle, and the specified percentage of the time that the
+output should be on. It cycles the output when the <code>go</code> input is on;
```

Comment:
Use `true` or `false`, not "on" as there is no such data type.
Also, it is odd to put the input to `false` and then this means the output is `true`. Is this the intent?